packages/block-editor/src/hooks/color.js

```diff
@@ -137,6 +139,19 @@ export function addSaveProps( props, blockType, attributes ) {
 	);
 	props.className = newClassName ? newClassName : undefined;
 
+	// Add "has-background-color-light" and "has-background-color-dark" classes,
+	// depending on the luminosity of the selected background color.
+	if ( backgroundColor ) {
+		props.className += tinycolor(
+			getColorObjectByAttributeValues(
+				wp.data.select( 'core/block-editor' ).getSettings().colors,
```

Comment:
Hi @aristath,

The save function or the addSaveProps hooks needs to be a pure function on the attributes. Given the same attributes, it must always return the same output.
If one uses wp.data.select(...) the function stops being pure e.g: even if the attributes don't change the result of the function may change.
So for example, if in one theme the "primary" color was white the class has-background-color-light would be added when the primary color was selected as the background. If later the theme changes primary color to black or the user switches themes, all the blocks using primary as background would be become invalid because the output of the save after the change would be different from what was stored in the markup.

Reply:
Thank you @jorgefilipecosta 
This was a weird & bad implementation, I was looking for ways to get the list of colors without over-complicating things.

This PR is no longer relevant so I'll close it, a far superior suggestion was made in https://github.com/WordPress/gutenberg/issues/24166#issuecomment-676050597 (similar to what you did for the pullquote styles)